lambdas/DumpSnippets/lambda_function.py

```diff
@@ -0,0 +1,77 @@
+import os
+import json
+import boto3
+import zipfile
+import base64
+
+# Initialization
+s3 = boto3.client('s3', 'us-west-2')
+try:
+    bn = os.environ['BucketName']
+except:
+    print 'Must specify "BucketName" env var!'
+    raise error
+try:
+    zb = os.environ['ZipsBucket']
+except KeyError:
+    zb = 'codesplain-zips'
+snippets_bucket = boto3.resource('s3').Bucket(bn)
+zips_bucket     = boto3.resource('s3').Bucket(zb)
+
+
+def zip_snippets(snippet_keys):
+    """Writes the contents of the snippets with given keys to a zip file,
+       returns the path to the zip."""
+    zip_path = '/tmp/dump.zip'
+    with zipfile.ZipFile(zip_path, 'w') as dump_zip:
```

Comment:
The time difference is unlikely to be noticed by the user. It takes a couple of seconds as is:
```bash
$ time curl -X POST https://kq6snf41mh.execute-api.us-west-2.amazonaws.com/BucketDump209/users/maryvilledev/dump
https://codesplain-zips.s3.amazonaws.com/maryvilledev/snippets.zip?AWSAccessKeyId=ASIAIJXIZ7K7EKJMXRHA&Expires=1495805067&x-amz-security-token=FQoDYXdzEBYaDG3zEDUSNyQ7yiSANiLzAT9zbovvM2H1XTi9MQ4wqESqMVqfRGWZMRbxeE3ilMka%2FVBCBSwOtv%2F6if2zRoSlAnA80yWnZcoA0Bioj7KBfLhWsOSTTxD%2FiaHNY9Iifg80mo%2B1uvSp9150xPM9g%2Fpnj4IBTuyMmsIlIMa4Sa8m4aXolrRhwoIV%2FH%2F6Se1y05mj6o%2Bkadvr5c1KgKXrRfM6RbzirJlgYzB4pEt0RjDGSjobgt0uIRn%2FzmDOuLFGgkjO4Zd5ie3uBfz3S7pXN9KliqZlGXXBkVBpqMQgWfKvGTk0WMa5fi0nAhtH33%2FAlI%2B2E0%2BrgkwG2mKXH8WS4s5rnFZGiSimvKDJBQ%3D%3D&Signature=uiKWis9FGK5DKxbM%2BuhEQA3ICQw%3D
real	0m2.696s
user	0m0.014s
sys	0m0.009s
```
I suspect the user would be willing to sacrifice a couple seconds of time in order to have more snippets.

As far as price is concerned, we are charged by [memory](https://aws.amazon.com/lambda/pricing/) as well as run time. Besides that, this lambda is unlikely to be used very frequently, so the price is not nearly as much concern for it as for, say, the `GetSnippetFromS3` lambda.